internal/services/mongocluster/mongo_cluster_resource.go

```diff
@@ -488,10 +488,6 @@ func (r MongoClusterResource) CustomizeDiff() sdk.ResourceFunc {
 					return fmt.Errorf("`administrator_username` is required when `create_mode` is %s", string(mongoclusters.CreateModeDefault))
 				}
 
-				if state.AdministratorPassword == "" {
-					return fmt.Errorf("`administrator_password` is required when `create_mode` is %s", string(mongoclusters.CreateModeDefault))
-				}
```

Comment:
This validation should probably be moved to the Create function instead of removed entirely

Reply:
Even if the validation is placed in the create function, this code will never be hit because administrator_username already has the constraint RequiredWith: []string{"administrator_password"}. Below is the possible scenarios.